### ISSUE TITLE
rails test directory param is ignored when running tests

### ISSUE THREAD

**Issue by jirihradil** [rails/rails#34662]
### Steps to reproduce
```
./bin/rails test test/models
```

### Expected behavior
Only tests for models should be executed.

### Actual behavior
All test are executed (including controllers and integration). When running `./bin/rake test:models`, it works as expected

### System configuration
**Rails version**: 5.0.7.1 

Version 5.0.7 worked fine.

**Ruby version**:
2.4.4

**Comment by y-yagi** [comment]
Thanks for your report. But I can't reproduce. Can you please provide a sample application that reproduces the issue?

**Comment by jirihradil** [comment]
OK, I tried to reproduce the issue on other machines as well and the problem is not in Rails. Sorry I haven't found it earlier.

The issue started by upgrading macOS to Mojave (10.14.2), and it seems to be a terminal or system-wide issue.

I found a workaround - if you append asterisk to the end of the directory path, the desired tests are running properly:

This **does not** work on macOS Mojave in Terminal:
```./bin/rails test test/models```

This **does** work:
```./bin/rails test test/models/*```
